src/bluesearch/entrypoint/database/add_es.py

```diff
@@ -0,0 +1,171 @@
+    """Yield an article mapping as a document to upload to Elasticsearch.
+
+    Parameters
+    ----------
+    inputs
+        Paths to the parsed files.
+    progress
+        Progress bar to update.
+
+    Yields
+    ------
+    dict[str, Any]
+        A document to upload to Elasticsearch.
+    """
+    for inp in inputs:
+        serialized = inp.read_text("utf-8")
+        article = Article.from_json(serialized)
+        doc = {
+            "_index": "articles",
+            "_id": article.uid,
+            "_source": {
+                "article_id": article.uid,
+                "authors": article.authors,
+                "title": article.title,
+                "abstract": article.abstract,
+                "pubmed_id": article.pubmed_id,
+                "pmc_id": article.pmc_id,
+                "doi": article.doi,
+            },
+        }
+        if progress:
+            progress.update(1)
+        yield doc
+
+
+def bulk_paragraphs(
+    inputs: Iterable[Path], progress: Optional[tqdm.std.tqdm] = None
+) -> Iterable[dict[str, Any]]:
+    """Yield a paragraph mapping as a document to upload to Elasticsearch.
+
+    Parameters
+    ----------
+    paragraph_mappings
+        The mappings of the paragraphs to upload.
+    progress
+        Progress bar to update.
+
+    Yields
+    ------
+    dict[str, Any]
+        A document to upload to Elasticsearch.
+    """
+    for inp in inputs:
+        serialized = inp.read_text("utf-8")
+        article = Article.from_json(serialized)
+        for ppos, (section, text) in enumerate(article.section_paragraphs):
+            doc = {
+                "_index": "paragraphs",
+                "_source": {
+                    "article_id": article.uid,
+                    "section_name": section,
+                    "text": text,
+                    "paragraph_id": ppos,
+                },
+            }
+            if progress:
+                progress.update(1)
+            yield doc
+
+
+def run(
+    client: Elasticsearch,
+    parsed_path: Path,
+) -> int:
+    """Add an entry to the database.
+
+    Parameter description and potential defaults are documented inside of the
+    `get_parser` function.
+    """
+    inputs: Iterable[Path]
+    if parsed_path.is_file():
+        inputs = [parsed_path]
+    elif parsed_path.is_dir():
+        inputs = sorted(parsed_path.glob("*.json"))
+    else:
+        raise ValueError(
+            "Argument 'parsed_path' should be a path to an existing file or directory!"
+        )
+
+    if len(inputs) == 0:
+        raise RuntimeWarning(f"No article was loaded from '{parsed_path}'!")
+
+    logger.info("Uploading articles to the database...")
+    progress = tqdm.tqdm(desc="Uploading articles", total=len(inputs), unit="articles")
+    resp = bulk(client, bulk_articles(inputs, progress))
+    logger.info(f"Uploaded {resp[0]} articles.")
+
+    if resp[0] == 0:
+        raise RuntimeWarning(f"No article was loaded from '{parsed_path}'!")
```

Comment:
Just for me to understand, when can this happen knowing that we have the `RuntimeWarning` when `len(inputs)=0`. Is it going to be when there is something wrong with `ElasticSearch` ? I am wondering if we could make the error message a bit more clear maybe, what do you think ?